mlrun/common/schemas/model_monitoring/model_endpoint_v2.py

```diff
@@ -0,0 +1,147 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import abc
+from typing import Optional
+
+from pydantic import BaseModel, Field, constr
+
+from ..object import ObjectKind, ObjectMetadata, ObjectSpec, ObjectStatus
+from .constants import (
+    PROJECT_PATTERN,
+    EndpointType,
+    ModelMonitoringMode,
+)
+from .model_endpoints import _mapping_attributes
+
+# TODO : replace ModelEndpoint
+
+
+class ModelEndpointParser(abc.ABC, BaseModel):
+    @classmethod
+    def json_parse_values(cls) -> list[str]:
+        return []
+
+    @classmethod
+    def from_flat_dict(cls, endpoint_dict: dict, json_parse_values: list = None):
+        """Create a `ModelEndpointMetadata` object from an endpoint dictionary
+
+        :param endpoint_dict:     Model endpoint dictionary.
+        :param json_parse_values: List of dictionary keys with a JSON string value that will be parsed into a
+                                  dictionary using json.loads().
+        """
+        if json_parse_values is None:
+            json_parse_values = cls.json_parse_values()
+
+        return _mapping_attributes(
+            model_class=cls,
+            flattened_dictionary=endpoint_dict,
+            json_parse_values=json_parse_values,
+        )
+
+
+class ModelEndpointV2Metadata(ObjectMetadata, ModelEndpointParser):
+    project: constr(regex=PROJECT_PATTERN)
+    endpoint_type: Optional[EndpointType] = EndpointType.NODE_EP.value
+
+
+class ModelEndpointV2Spec(ObjectSpec, ModelEndpointParser):
+    model_uid: Optional[str] = ""
+    model_name: Optional[str] = ""
+    model_tag: Optional[str] = ""
+    model_class: Optional[str] = ""
+    function_name: Optional[str] = ""
+    function_uid: Optional[str] = ""
+    feature_names: Optional[list[str]] = []
+    label_names: Optional[list[str]] = []
+
+
+class ModelEndpointV2Status(ObjectStatus, ModelEndpointParser):
+    state: Optional[str] = "unknown"  # will be updated according to the function state
+    first_request: Optional[str] = ""
+    children: Optional[list[str]] = []
+    children_uids: Optional[list[str]] = []
+    monitoring_feature_set_uri: Optional[str] = ""
+    monitoring_mode: Optional[ModelMonitoringMode] = ModelMonitoringMode.disabled.value
+    function_uri: Optional[str] = ""  # <project_name>/<function_name>:<tag>
+    model_uri: Optional[str] = ""
+
+    # operative
+    last_request: Optional[str] = ""
+    drift_status: Optional[str] = ""
+    avg_latency: Optional[float] = None
+    error_count: Optional[int] = None
```

Comment:
Shouldn't it be` 0`?  Until now, the default value was `0`, which makes sense since that’s the number displayed in the UI if no error occurs
```suggestion
    error_count: Optional[int] = 0
```